src/utils/normalizeUrl.ts

```diff
@@ -0,0 +1,11 @@
+export function normalizeUrl(url?: string): string | undefined {
+  if (!url || url.trim().length === 0) {
```

Comment:
The `.trim()` may be a bit overly defensive, but since we're also normalizing user-provided values, I thought it might be good to check.